### PR TITLE
README and macOS versions update

### PR DESCRIPTION
**Pull request by DrJosh9000**
- EOL platforms may be dropped from new minor releases
- Fix lies in the macOS versions table (e.g. Go 1.12 was the last Go to support Yosemite - not 1.13. Go 1.13 required El Capitan. This also impacts agent versions, etc) and add links
- Add linebreaks
- Fix Markdown lints
- Refresh `buildkite-agent --help` output
- Update copyright year
- Change `[here]` to something else
- Change telemetry features link

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -126,13 +154,19 @@ support guidance).
...
 
 ### Operating System Support
 
-We currently provide support for running the Buildkite Agent on the following operating
-systems. Future major releases may drop support for old operating systems. The agent
-binary is fairly portable and should run out of the box on most UNIX like systems.
+We currently provide support for running the Buildkite Agent on the following
+operating systems. Future _minor_ releases may drop support for end-of-life
+operating systems (typically as they become unsupported by the latest stable Go
+release).
```
Comment (triarius) [inline README.md:166]: Yes!

File: README.md [hunk 2]
```diff
@@ -1,115 +1,143 @@
...
+  Ubuntu 20.04
+- 3.45-ubuntu-20.04, tracks bugfix updates in version 3.45 installed in Ubuntu
+  20.04
+- 3.45.6-ubuntu-20.04, tracks the exact version installed in Ubuntu 20.04
 
 #### Supported operating systems
 
-- Alpine 3.12
+- Alpine 3.18
 - Ubuntu 18.04 LTS (x86_64), supported to end of life for 18.04
```
Comment (triarius) [inline README.md:71]: I would like us to make a decision to change this to "supported to end of standard support for 18.04"

EoL is 10 years, standard support is 5 years.

**Review by triarius — APPROVED** [review]
LGTM. I approve of the changes to the support policy. Do we need to make an announcement about them?

**Comment by DrJosh9000** [comment]
> Do we need to make an announcement about them?

I think so.

Whether we do or not, this change at least makes the README more honest about the _status quo_. Looking back to the last few times we did a Go bump that removed old OS support (Go 1.17 in #1549 dropped macOS 10.12 support, Go 1.18 in #1603 dropped Linux kernels pre 2.6.32, Go 1.19 in #1774 dropped FreeBSD 11.x[^freebsd]) they happened without much fanfare or probably without us even realising. 

Still, I'll add a callout to the v3.54 changelog and release notes. Maybe the docs too.

[^freebsd]: Mentioned in the 1.18 notes, but not the 1.19 notes, so I wonder if even the Go team just forgot: https://go.dev/doc/go1.18#freebsd

File: README.md (hunk 2 above)
Comment (DrJosh9000) [reply README.md]: SGTM!

File: README.md [hunk 3]
```diff
@@ -142,17 +176,18 @@ binary is fairly portable and should run out of the box on most UNIX like system
...
-  - 10.15
-  - 11
+  - 10.13 (High Sierra)
+  - 10.14 (Mojave)
+  - 10.15 (Catalina)
+  - 11 (Big Sur)
+  - 12 (Monterey)
+  - 13 (Ventura)
 - Windows Server
   - 2012
```
Comment (triarius) [inline README.md:186]: Very much looking forward to Oct 10, so we can drop this too.

File: README.md [hunk 4]
```diff
@@ -142,17 +176,18 @@ binary is fairly portable and should run out of the box on most UNIX like system
...
   - 2016
   - 2019
```
Comment (triarius) [inline README.md:188]: There's a Windows Server 2022. I've never run an agent on it, though.

**Review by triarius — APPROVED** [review]

**Pull request merged**